### DISCUSSION TITLE
Add content to existing model templates -> how?

### DISCUSSION THREAD

**Discussion by MalfuncEddie** [netbox-community/netbox#12780] [Q&A] [answered]
Hi,

I am probably missing something basic while reading the documentation. In the getting started section it states "Add content to existing model templates" 
Am I correct that this means I can add  it to a core model?
Because this conflicts with "Modify core models is not allowed"

eg:

I have a model and I assume that I get all this working (with the very good and impressive documentation). I've got create edit table views/templates just like de documentation. 

The model is a "child" of the core model VM ( VM has one or more the new models). The documentation  says I can inject a template in a core model. So I should be able to inject a list of the models in the VM template? 

I was wondering if this is described somewhere? The plugin tutorial seems to contain the plugin to it's models it self.
What if I want to add the AccessList model from the tutorial to a device and make it visible that there is a AccessList in de device view?  is this possible?

Sources:
https://docs.netbox.dev/en/stable/plugins/
and 
https://github.com/netbox-community/netbox-plugin-tutorial

**Comment by candlerb** [comment]
> Am I correct that this means I can add it to a core model?

No. It means you can add content to the *views* of an existing core model - such as data from your plugin's own models, which relates to an existing model.  But you cannot modify the core models themselves.

For example: if you have a Foo model, which contains a reference to Device, then when viewing a Device you can also show the related Foos by adding content into the Device view.  But this does not change the Device model.

**Reply by MalfuncEddie** [reply]
So if my "plugin model" has a remote if to a core model I can list the "plugin objects" in the core model view?

Your example is what I am looking for in the documentation but this does not seem present in the documentation tutorial?

**Reply by candlerb** [reply, marked as answer]
I think what you need is here: https://docs.netbox.dev/en/stable/plugins/development/views/#extending-core-views

**Reply by MalfuncEddie** [reply]
Thank you so much!
